content/posts/blog/2024-03-13-image-and-distribution-1-1.md

```diff
@@ -0,0 +1,209 @@
+      "size": 0
+    }
+  ],
+  "subject": {
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "digest": "sha256:21edd7d11800e94bae9f4...",
+    "size": 1769
+  },
+  "annotations": {
+    "org.opencontainers.image.created": "2024-03-02T15:16:17Z",
+    "org.opencontainers.image.description": "Example artifact for image sha256:21ed..."
+  }
+}
+```
+
+# Referrers API
+
+In the distribution-spec, the counterpart to the `subject` field is the the new [referrers API][distribution-referrers]: `GET /v2/<name>/referrers/<digest>`.
+This API returns an OCI Index listing the manifests where the `subject` digest matches the requested `<digest>` value.
+Each of the descriptors in the response also include the `artifactType` and `annotation` values from the referenced manifest.
+Including those values allows tooling to select their desired artifact from the OCI Index the same way runtimes use the platform to select an image to run from a multi-platform Index.
+
+The referrers response to `GET /v2/<name>/referrers/sha256:21edd7d11800e94bae9f4...` may look like:
+
+```jsonc
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:9e6569b15e5ed981334003eb8...",
+      "size": 724,
+      "annotations": {
+        "org.opencontainers.image.created": "2024-03-02T15:16:17Z",
+        "org.opencontainers.image.description": "Example artifact for image sha256:21ed..."
+      },
+      "artifactType": "application/example"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:e40ce69ad428e47eb72ea76d9...",
+      "size": 701,
+      "annotations": {
+        "org.opencontainers.image.created": "2024-03-03T15:15:15Z",
+        "org.opencontainers.image.description": "Another example artifact for same image"
+      },
+      "artifactType": "application/example2"
+    }
+  ]
+}
+```
+
+# Support For Existing Registries
+
+The design of the changes to the image and distribution specifications assume that it will take time for registries to upgrade.
+To facilitate this transition period, changes to the image spec added new fields which should be ignored by registries without generating errors or modifying the content of the manifest.
+Clients pushing a manifest with a `subject` field and querying for referrers have an additional responsibility to manage and use a [fallback tagged Index when the registry does not support the referrers API][distribution-referrers-fallback].
+When a registry supports the referrers API, it will respond to pushing a manifest with the `subject` with the header `OCI-Subject` set to the digest in the subject field.
+When querying for referrers, a registry will respond with a `404` or similar error code if the API is not supported (when no referrers are found, an Index with an empty manifest list would be returned).
```

Comment:
Language a little confusing ... are we mixing the cases of non-support and not-found?

Reply:
I did some rewording of these last two lines. Let me know if that helps.